contracts/SupplyChain.sol

```diff
@@ -1,19 +1,34 @@
-pragma solidity 0.5.0;
+pragma solidity 0.5.7;
 
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
 import "./Package.sol";
 
 
-contract SupplyChain {
+contract SupplyChain is Ownable {
 
     mapping(bytes => Package) private packages;
+    mapping(address => bool) private producers;
```

Comment:
You can move all producer related things to separate contract and then inheritance from it. It would be much cleaner.